### PR TITLE
jv-work-with-file

### PR DESCRIPTION
**Pull request by VolodymyrDavidov**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,71 @@
...
+    private static final String COMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String string = readFromFile(fromFileName);
+        String result = resultOfDataReading(string);
+        recordToFile(result,toFileName);
+    }
+
+    private String readFromFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/WorkWithFile.java:22]: do not use such naming(stringBuilder), how to guess what it stores?

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,71 @@
...
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String string = readFromFile(fromFileName);
+        String result = resultOfDataReading(string);
+        recordToFile(result,toFileName);
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/WorkWithFile.java:18]:
```suggestion
        recordToFile(result, toFileName);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,71 @@
...
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName));
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                stringBuilder.append(line).append(COMA);
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
+            throw new RuntimeException("This file cannot be read ", e);
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/WorkWithFile.java:31]:
```suggestion
            throw new RuntimeException("Can't read data from the file: " + fromFileName, e);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,71 @@
...
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/WorkWithFile.java:12]: The name of the constant should not be the same as the value being stored

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,71 @@
...
+            bufferedReader.close();
+        } catch (IOException e) {
+            throw new RuntimeException("This file cannot be read ", e);
+        }
+        return stringBuilder.toString();
+    }
+
+    private String resultOfDataReading(String dataOfFile) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String[] dataArray = dataOfFile.split(",");
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/WorkWithFile.java:38]:
```suggestion
        String[] dataArray = dataOfFile.split(COMMA);
```

**Review by Ivan95kos — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String DEFAULT_SUPPLY = "supply";
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:10]: why default? it denotes operation, better call it OPERATION_SUPPLY

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,7 +1,71 @@
...
+        int result = amountOfSupplies - amountOfBuy;
+        resultString.append(DEFAULT_SUPPLY).append(COMA).append(amountOfSupplies)
+                .append(System.lineSeparator()).append(DEFAULT_BUY).append(COMA)
+                .append(amountOfBuy).append(System.lineSeparator())
+                .append(DEFAULT_RESULT).append(COMA).append(result);
+
+        return resultString.toString();
+    }
 
+    private void recordToFile(String stringForWritingToFile, String toFileName) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:62]:
```suggestion
    private void writeToFile(String content, String toFileName) {
```
recording is for audio

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 8]
```diff
@@ -1,7 +1,71 @@
...
+                dataToString.append(line).append(COMA);
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file: " + fromFileName, e);
+        }
+        return dataToString.toString();
+    }
+
+    private String resultOfDataReading(String dataOfFile) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:36]:
```suggestion
    private String generateReport(String fileData) {
```
method names should be verbs, save nouns for class names :)

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 9]
```diff
@@ -1,7 +1,71 @@
...
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String DEFAULT_SUPPLY = "supply";
+    private static final String DEFAULT_BUY = "buy";
+    private static final String DEFAULT_RESULT = "result";
+    private static final String COMA = ",";
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:13]:
```suggestion
    private static final String DELIMITER = ",";
```
could be SEPARATOR too
and it's `comma` just FYI

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 10]
```diff
@@ -1,7 +1,71 @@
...
+                .append(System.lineSeparator()).append(DEFAULT_BUY).append(COMA)
+                .append(amountOfBuy).append(System.lineSeparator())
+                .append(DEFAULT_RESULT).append(COMA).append(result);
+
+        return resultString.toString();
+    }
 
+    private void recordToFile(String stringForWritingToFile, String toFileName) {
+        try {
+            BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName));
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:64]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#try-with-resources

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 11]
```diff
@@ -1,7 +1,70 @@
...
     public void getStatistic(String fromFileName, String toFileName) {
+        String string = readFromFile(fromFileName);
+        String result = generateReport(string);
+        writeToFile(result, toFileName);
+    }
+
+    private String readFromFile(String fromFileName) {
+        StringBuilder dataToString = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName));
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:24]: Please, use try-with-resources here
https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#try-with-resources

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 12]
```diff
@@ -1,7 +1,68 @@
...
+        String result = generateReport(string);
+        writeToFile(result, toFileName);
+    }
+
+    private String readFromFile(String fromFileName) {
+        StringBuilder dataToString = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                dataToString.append(line).append(SEPARATOR);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:26]:
```suggestion
                dataToString.append(line).append(System.lineSeparator());
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 13]
```diff
@@ -1,7 +1,68 @@
...
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file: " + fromFileName, e);
+        }
+        return dataToString.toString();
+    }
+
+    private String generateReport(String fileData) {
+        StringBuilder resultString = new StringBuilder();
+        String[] dataArray = fileData.split(SEPARATOR);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:36]:
```suggestion
        String[] dataArray = fileData.split(System.lineSeparator());
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 14]
```diff
@@ -1,7 +1,68 @@
...
+                String value = dataArray[i + 1];
+                amountOfSupplies = amountOfSupplies + Integer.parseInt(value);
+                i++;
+            }
+            if (dataArray[i].equals(OPERATION_BUY)) {
+                String value = dataArray[i + 1];
+                amountOfBuy = amountOfBuy + Integer.parseInt(value);
+                i++;
+            }
+        }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/WorkWithFile.java:50]: 1. use for each
2. inside a loop, split by SEPARATOR
3. retrieve amount and operation type by indexes which are class constants

**Review by okuzan — CHANGES_REQUESTED** [review]
Generating report logic can be refactored to be less error-prone. It's better to rely on constants, to have a clear separation between lines, and not transform input into one long command line

**Pull request closed without merging**